src/get-default-blueprint-name-override.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
```

Comment:
I see the unit test, but we should have a test showing this really working.

Reply:
Will add test case in the install integration test

Reply:
added

```
it('can install an addon with a custom blueprint that does not match package name', async function() {
    let addonContainingBlueprint = 'test/fixtures/blueprint/addon/default-blueprint-different-than-name';
    // In the package.json, the package version value will be 'file:../legacy-blueprint'
    let copiedAddonPackagePath = '../legacy-blueprint';

    let {
      status
    } = await merge({
      fixturesPath: 'test/fixtures/blueprint/addon/legacy-app/local/no-addon',
      commitMessage: 'my-app',
      addon: copiedAddonPackagePath,
      async beforeMerge() {
        await initBlueprint({
          fixturesPath: addonContainingBlueprint,
          resolvedFrom: tmpPath,
          relativeDir: copiedAddonPackagePath
        });

        await spawn('npm', ['install'], { cwd: tmpPath });
      },
      async afterMerge() {
        await fs.remove(path.join(tmpPath, 'package-lock.json'));
      }
    });

    fixtureCompare({
      mergeFixtures: 'test/fixtures/blueprint/addon/legacy-app/merge/my-app-install-blueprint-different-than-name'
    });

    assertNoStaged(status);
  });
```